external-api/src/repositories/estore/estore-repo.ts

```diff
@@ -0,0 +1,80 @@
+import { InsertOneResult, ObjectId, UpdateFilter, UpdateResult, WithId, WithoutId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS } from '@ukef/dtfs2-common';
+import { getCollection } from '../../database';
+
+/**
+ * Finds a document in the CRON_JOB_LOGS collection based on the provided deal ID.
+ *
+ * @param {ObjectId} dealId - The unique identifier for the deal.
+ *
+ * @returns {Promise<WithId<ObjectId> | WithoutId<unknown> | null>} - A promise that resolves to the found document, or `null` if no document is found.
+ *
+ * @example
+ * const dealId = new ObjectId('507f1f77bcf86cd799439011');
+ *
+ * find(dealId)
+ *   .then((document) => console.log('Document found', document))
+ *   .catch((error) => console.error('Find operation failed', error));
+ */
+export const findByDealId = async (dealId: ObjectId): Promise<WithId<object> | null> => {
+  try {
+    const collection = await getCollection(MONGO_DB_COLLECTIONS.CRON_JOB_LOGS);
+
+    return collection.findOne({ 'payload.dealId': { $eq: dealId } });
+  } catch (error) {
+    console.error('An error occurred while finding deal %s in %s collection.', dealId, MONGO_DB_COLLECTIONS.CRON_JOB_LOGS);
+    return null;
+  }
+};
+
+/**
+ * Inserts a document into the CRON_JOB_LOGS collection.
+ *
+ * @param {object} document - The document to be inserted.
+ *
+ * @returns {Promise<InsertOneResult | boolean>} - A promise that resolves to the result of the insert operation, or `false` if an error occurs.
+ *
+ * @example
+ * const document = { dealId: '507f1f77bcf86cd799439011', status: 'new' };
+ *
+ * insert(document)
+ *   .then((result) => console.log('Insert successful', result))
+ *   .catch((error) => console.error('Insert failed', error));
+ */
+export const insertOne = async (document: object): Promise<InsertOneResult | boolean> => {
```

Comment:
I think the type of `document` here can be the `Document` type (imported from `mongodb`). The result type can also be narrowed to be `Promise<InsertOneResult | false>`